pkg/k8s/watchers/cilium_network_policy.go

```diff
@@ -160,22 +244,25 @@ func (k *K8sWatcher) onUpsertCNP(
 	// See https://github.com/cilium/cilium/blob/27fee207f5422c95479422162e9ea0d2f2b6c770/pkg/policy/api/ingress.go#L112-L134
 	cnpCpy := cnp.DeepCopy()
 
+	translatedCNP := resolveCIDRGroupRef(cnpCpy, cidrGroupCache)
+
 	var err error
 	if ok {
-		err = k.updateCiliumNetworkPolicyV2(cs, oldCNP, cnpCpy, initialRecvTime)
+		err = k.updateCiliumNetworkPolicyV2(cs, oldCNP, translatedCNP, initialRecvTime)
 	} else {
-		err = k.addCiliumNetworkPolicyV2(cs, cnpCpy, initialRecvTime)
+		err = k.addCiliumNetworkPolicyV2(cs, translatedCNP, initialRecvTime)
 	}
 	if err == nil {
-		cache[key] = cnpCpy
+		// We keep in the cache the version of the cnp ahead of any CIDRGroupRef translation
```

Comment:
Can we clarify this to say "The CNP cache contains the original form of the CNP (pre-translation of CIDRGroupRef translation)"? I think it also makes more sense to move the comment to where the `cnpCache` is initialized.